File: Math/2756_다트.cpp
#include <iostream>
#include <cmath>

using namespace std;

int get_score(double x, double y) {
	int score = 0;
	double len = sqrt((x * x) + (y * y));

	if (len <= 3)
		score += 100;
	else if (len <= 6)
		score += 80;
	else if (len <= 9)
		score += 60;
	else if (len <= 12)
		score += 40;
	else if (len <= 15)
		score += 20;

	return score;
}

int main(void) {
	//ÃÊ±âÈ­
	ios::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	int T;
	
	cin >> T;
	while (T--) {
		int player_1 = 0;
		int player_2 = 0;

		for (int i = 0; i < 6; ++i) {
			double x, y;
			cin >> x >> y;

			if (i < 3) 
				player_1 += get_score(x, y);
			else
				player_2 += get_score(x, y);
		}

		cout << "SCORE: " << player_1 << " to " << player_2 << ", ";
		if (player_1 > player_2)
			cout << "PLAYER 1 WINS.\n";
		else if (player_1 < player_2)
			cout << "PLAYER 2 WINS.\n";
		else
			cout << "TIE.\n";
	}
	return 0;
}

File: Math/2756_다트.py
#입출력
from sys import stdin

#입력 
input = stdin.readline

T=int(input())

for _ in range(T):
    #좌표 입력
    points=list(map(float,input().split(' ')))
    
    player1 = 0
    player2 = 0
    winner = None

    #2개씩 건너뛰며
    for i in range(0, 12, 2):
        score = 0
        dist = points[i]**2 + points[i+1]**2
       
        if dist <= 9:
            score += 100
        elif 9 < dist <= 36:
            score += 80
        elif 36 < dist <= 81:
            score += 60
        elif 81 < dist <= 144:
            score += 40
        elif 144 < dist <= 225:
            score += 20

        if 0 <= i <= 4:
            player1 += score
        else:
            player2 += score

    if player1 > player2:
        winner = "PLAYER 1 WINS"
    elif player1 < player2:
        winner = "PLAYER 2 WINS"
    else:
        winner = "TIE"

    print(f'SCORE: {player1} to {player2}, {winner}.')
   
